### ISSUE TITLE
quickquasars: add Lorentzian 

### ISSUE THREAD

**Issue by londumas** [desihub/desisim#441]
Figure 4 of Zarrouk et al 2018 (https://arxiv.org/pdf/1801.03062.pdf) shows that the redshift error are Lorentzian.

**Comment by andreufont** [comment]
Is the argument that FoG is Lorentzian, or that the redshift errors are? 

The modelling of FoG can be informed from Nbody simulations, I guess. 

If we decided that either redshift errors or FoG are better modelled by a Lorentzian, this should be trivial to change.

**Comment by londumas** [comment]
@alxogm and @andreufont, here is what I had in mind:
```
import scipy as sp
import scipy.stats
import scipy.constants
def lorentzian(nb, mean, var):
    return sp.stats.cauchy.rvs(loc=mean,scale=var,size=nb)
def gaussian(nb, mean, var):
    return sp.stats.norm.rvs(loc=mean,scale=var,size=nb)

mean = 0.
var = 200.
dv = lorentzian(nb, mean, var)
dv = gaussian(nb, mean, var)
```

In the following plot I compare these two functions to the difference between two estimator of redshift
in DR12Q.
I am planning to do the same plot with two redshift with the same estimator, but on two different epochs.

![deltav_distribution](https://user-images.githubusercontent.com/3166436/48079871-8c0a2d80-e1a9-11e8-8bde-b1e05167c60f.png)

![deltav_distribution_zoom](https://user-images.githubusercontent.com/3166436/48079873-8c0a2d80-e1a9-11e8-9033-67e4a4d57d24.png)

**Comment by alxogm** [comment]
Hello @londumas, thank you for the plot. This is similar to what the plots I was doing when I thought of truncating the Cauchy distribution in the mod_cauchy function... I should've showed it instead of trying to explain it ...  I didn't have the DR12Q data to compare directly but by eye I thought just forcing the samples to be inside some limits (delta v<3000) could be sufficient.  Actually in the mock we are adding a gaussian with sigma=150(for the fingers of god effect),  and the Lorentzian. We can add another gaussian and choose the sigma to match the data.

Is that DR12Q data in NERSC? or is this plot in desi_QA?  I could compare how the mock looks like in comparison  and play with the different parameters, maybe more than one Lorentzian, with samples within different ranges? 

Thanks

**Comment by andreufont** [comment]
I have the feeling that we are spending a lot of time on this issue, and I'm not convinced it should be our priority at this point... I asked Helion to open these issues on github so that we don't forget, but I think solving the DLA issues is more important. 

However, if you think we can finish this up soon, then we should push this and get it out of the way. But remember that at the end what we want is to mimic the redshift fitting errors in DESI, that might be somewhat different than in BOSS/eBOSS. So I don't think we'll be able to fully finish this up until we have SV data. 

By the way, I started #447 that is marginally related, and not a priority either.

**Comment by londumas** [comment]
@andreufont, I agree, let's not invest too much time with that. However, it is important for the mocks to have it sooner than later because this jas a big effect on the 3d cross-correlation. Way more than metals and DLA.
Let's correct the current code and merge https://github.com/desihub/desisim/commit/7098e8261feb443c4c522e6faf561cca5fcf44b8#commitcomment-31183244.
@alxogm, do you want to take care of this? In `mod_cauchy` you have a loop running on `nsamples`, this is very slow. Let's use `sp.stats.cauchy.rvs(loc=mean,scale=var,size=nb)` with nb three times larger than `nsamples` and thus take only the `Delta v` that are within `[-3000,3000]`.
Do you want me to take care of this or you do it?

**Comment by alxogm** [comment]
@londumas Yes I agree the modified function can be optimized as you suggest. Also agree on not spending much more time on it. I'm changing the code now and will  make a PR 

@andreufont  If I understand right issue #447 is change a line in the code, so I'll include it in this PR and you let me know if that is what was needed.

Thanks 
Alma

**Comment by alxogm** [comment]
Hi @londumas I made the changes. https://github.com/desihub/desisim/compare/qquasar_addfeature I'll make a PR when tests passes, if you agree.

**Comment by londumas** [comment]
Hi @alxogm, Thank you for the work, we can move on and I be happy to test the results on the next iteration of the mocks.

**Comment by londumas** [comment]
@alxogm, Seems to be fixed. Can we close?
